### PR TITLE
[Manual Taxes M2] Improvements

### LINKED ISSUE #10646: [Manual Taxes M2] Filter tax rates without addresses

### PR DESCRIPTION
**Pull request by toupper**
Closes: #10646 

## Description

With this PR we want to add some improvements to the Manual Taxes Milestone 2:

- We filter those tax rates that don't have a location from the list.
- We change the bottom text for the empty state.
- We support older Woo versions when converting the TaxRate to an address
- We improve some comments
- We hide the Set New Tax Rate if taxes are based on shop address
- We refresh the tax rates after the wp-admin web view is closed to refresh values

### Hide Set New Tax Rate in Order Details if taxes are based on shop address

With tax based on shop address (You can set that in wp-admin/admin.php?page=wc-settings&tab=tax)

1. Go to Orders
2. Tap on + to select a new order
3. Check that the Set New Tax Rate row is hidden

Try changing the setting to customer shipping address or billing address. The row should be there.

### Refresh after web view is closed

https://github.com/woocommerce/woocommerce-ios/assets/1864060/d9f52104-1ef7-4829-b36b-50a73265260c

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by joshheald** [comment]
> * We filter those tax rates that don't have a location from the list.

Why? Is it not valid to select these tax rates for some reason, or will they have been applied automatically and not be removable?

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorViewModel.swift [hunk 1]
```diff
@@ -175,3 +197,20 @@ extension NewTaxRateSelectorViewModel {
         syncState = taxRateViewModels.isNotEmpty ? .results: .empty
     }
 }
+
+extension NewTaxRateSelectorViewModel {
+    enum Localization {
+        static let bottomNoticeResultsSectionTitle = NSLocalizedString("Can’t find the rate you’re looking for?",
+                                                                         comment: "Text to prompt the user to edit tax rates in the web")
+        static let bottomNoticeEmptySectionTitle = NSLocalizedString("You don't have any tax rate with a location.",
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorViewModel.swift]:
```suggestion
        static let bottomNoticeEmptySectionTitle = NSLocalizedString("You don't have any tax rates with a location.",
```

**Review by joshheald — APPROVED** [review]
Works as described – one tiny string correction suggested

**Comment by toupper** [comment]
> Why? Is it not valid to select these tax rates for some reason, or will they have been applied automatically and not be removable?

The target of this second milestone is to use the tax rate location (city, state, country and/or postcode) to the order customer's address, so the tax rate can take effect (this only happens if the user sets the taxes based on billing or shipping address). If the tax rate hasn't any location we can't set it to the order customer's address, and the tax rate won't be applied. Because of this, we filter them, otherwise, it would be confusing because nothing would happen after selecting such a tax rate.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorViewModel.swift (hunk 1 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorViewModel.swift]: It's great to have a native speaker in the team, thanks! Changed in 787d8789ef48f7ee509eae4a70a22c95a507ef0f

**Pull request merged**